web/src/components/product/RegistrationExtension.tsx

```diff
@@ -136,24 +141,18 @@ export default function RegistrationExtension({
           <RegisteredExtensionStatus registrationCode={registrationData.registrationCode} />
         )}
         {!isRegistered && extension.available && !extension.free && (
-          <Form id={`register-form-${extension.id}-${extension.version}`} onSubmit={submit}>
+          <Form id={formId} onSubmit={submit}>
             {/* // TRANSLATORS: input field label */}
-            <FormGroup label={_("Registration code")}>
+            <FormGroup fieldId={inputId} label={_("Registration code")}>
               <RegistrationCodeInput
                 isDisabled={loading}
-                id={`input-reg-code-${extension.id}-${extension.version}`}
+                id={inputId}
                 value={regCode}
                 onChange={(_, v) => setRegCode(v)}
               />
             </FormGroup>
             <ActionGroup>
-              <Button
-                id={`register-button-${extension.id}-${extension.version}`}
```

Comment:
For consistency with other ids, I moved the id into a variable. That said, I’m not sure if it’s still necessary since it seems like it was originally added for testing purposes, but I have updated the test to avoid relying on ids. I’ve kept it for now to be safe, but it might be worth revisiting and potentially removing later.